zcash_client_sqlite/src/wallet/init/migrations/add_transaction_views.rs

```diff
@@ -85,20 +85,22 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
         while let Some(row) = tx_rows.next()? {
             let id_tx: i64 = row.get(0)?;
             let tx_bytes: Option<Vec<u8>> = row.get(1)?;
-            let h: u32 = row.get(2)?;
-            let block_height = BlockHeight::from(h);
+            let block_height: Option<u32> = row.get(2)?;
+            let block_height = block_height.map(BlockHeight::from);
 
             // If only transaction metadata has been stored, and not transaction data, the fee
             // information will eventually be set when the full transaction data is inserted.
-            if let Some(b) = tx_bytes {
-                let tx =
-                    Transaction::read(&b[..], BranchId::for_height(&self.params, block_height))
-                        .map_err(|e| {
-                            WalletMigrationError::CorruptedData(format!(
-                                "Parsing failed for transaction {:?}: {:?}",
-                                id_tx, e
-                            ))
-                        })?;
+            if let Some((tx_bytes, block_height)) = tx_bytes.zip(block_height) {
+                let tx = Transaction::read(
+                    &tx_bytes[..],
+                    BranchId::for_height(&self.params, block_height),
```

Comment:
The problem here is that we don't necessarily know the block height in order to figure out the consensus branch ID. But doing so only matters for v4 transactions; v5 transactions ignore the `BranchId` passed in and use the one stored internally. So at a minimum we can reliably parse v5 transactions by passing a dummy value, and then if the resulting transaction is v4 and `block_height.is_none()` then we could filter those out as not reliable.

But my second question is, does anything that `tx` is used for below need the consensus branch ID? If it doesn't affect the `Transaction::fee_paid` result, then we could just not read `block_height` from the database at all and use a dummy value all the time (like we do in a few places in `zcashd`).

Reply:
Oh, that's a good point; we're not doing anything with consensus_branch_id, so we can just always use Nu5, since all we actually want is the prevouts and value balances.